server/api.go

```diff
@@ -148,7 +148,7 @@ func (p *Plugin) setFilePermissions(fileID, userID, permission string) error {
 		return errors.Wrap(postError, "error when retrieving post for file")
 	}
 
-	if post.UserId != userID {
+	if !skipFileOwnerCheck && post.UserId != userID {
```

Comment:
Should we have checked for `fileInfo.userID` as in your other PR?

Reply:
actually `fileInfo` doesn't have a userID field. ref - https://pkg.go.dev/github.com/mattermost/mattermost-server/v5/model#FileInfo.
In the web app, the `FileInfo` type did have a `userID` field for the newly created attachment only.